packetbeat/sniffer/sniffer.go

```diff
@@ -175,20 +184,27 @@ func validateAfPacketConfig(cfg *config.InterfacesConfig) error {
 // Run opens the sniffing device and processes packets being read from that device.
 // Worker instances are instantiated as needed.
 func (s *Sniffer) Run() error {
-	var (
-		defaultRoute chan string
-		refresh      chan struct{}
-	)
-	if s.followDefault {
-		s.done = make(chan struct{})
-		defaultRoute = make(chan string)
-		refresh = make(chan struct{}, 1)
-		go s.pollDefaultRoute(defaultRoute, refresh)
-	}
-	if defaultRoute == nil {
-		return s.sniffStatic(s.device)
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
With the sniffer also utilizing the `done` channel as a shutdown signal, I would consider a future refactoring to combine the shutdown signalling into a single `context.Context`. Like having a ctx that comes down from the beater package that gets cancelled on shutdown and then pass that into this errgroup as its parent.